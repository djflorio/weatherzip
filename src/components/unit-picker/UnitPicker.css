.unit-picker {
  background-color: #fff;
  margin-bottom: 10px;
  border-radius: 5px;
  box-shadow: 0 0 2px #222;
  animation: bounceIn 1s 0.1s both;
  overflow: hidden;
  cursor: pointer;
}

.unit-picker__choice {
  display: inline-block;
  width: 50%;
  font-size: 1.3em;
  padding: 8px 0;
  color: #aaa;
}

.unit-picker__choice--active {
  background-color: rgb(30, 133, 180);
  color: #fff;
}