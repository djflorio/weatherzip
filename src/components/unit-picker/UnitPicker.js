// Third party
import React from 'react';
import PropTypes from 'prop-types';

// Assets
import './UnitPicker.css';


const UnitPicker = (props) => (
  <div className="unit-picker">
    <div
      className={"unit-picker__choice" + (props.units === "imperial" ? " unit-picker__choice--active" : "")}
      onClick={() => props.onChange("imperial")}>
      &deg;F
    </div>
    <div className={"unit-picker__choice" + (props.units === "metric" ? " unit-picker__choice--active" : "")}
      onClick={() => props.onChange("metric")}>
      &deg;C
    </div>
  </div>
);

UnitPicker.propTypes = {
  units: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
}

export default UnitPicker;