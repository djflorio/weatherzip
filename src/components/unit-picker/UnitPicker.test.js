import UnitPicker from './UnitPicker';

it("renders correctly with imperial", () => {
  const wrapper = shallow(
    <UnitPicker
      units="imperial"
      onChange={() => {}}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it("renders correctly with metric", () => {
  const wrapper = shallow(
    <UnitPicker
      units="metric"
      onChange={() => {}}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it("calls onChange with correct value", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <UnitPicker
      units="imperial"
      onChange={spy}
    />
  );

  wrapper
    .find(".unit-picker__choice")
    .first()
    .simulate("click");

  expect(spy.calledWith("imperial"));

  wrapper
    .find(".unit-picker__choice")
    .at(1)
    .simulate("click");

  expect(spy.calledWith("metric"));
});