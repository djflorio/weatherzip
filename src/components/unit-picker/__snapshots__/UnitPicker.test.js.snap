// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with imperial 1`] = `
<div
  className="unit-picker"
>
  <div
    className="unit-picker__choice unit-picker__choice--active"
    onClick={[Function]}
  >
    째F
  </div>
  <div
    className="unit-picker__choice"
    onClick={[Function]}
  >
    째C
  </div>
</div>
`;

exports[`renders correctly with metric 1`] = `
<div
  className="unit-picker"
>
  <div
    className="unit-picker__choice"
    onClick={[Function]}
  >
    째F
  </div>
  <div
    className="unit-picker__choice unit-picker__choice--active"
    onClick={[Function]}
  >
    째C
  </div>
</div>
`;
